build_openroad.sh

```diff
@@ -168,6 +168,7 @@ elif [ "$build_method" == "LOCAL" ]; then
   cmake -B tools/build/LSOracle tools/LSOracle -D CMAKE_BUILD_TYPE=RELEASE -D YOSYS_INCLUDE_DIR=$(pwd)/tools/yosys -D YOSYS_PLUGIN=ON
   $NICE cmake --build tools/build/LSOracle -j$PROC
   mkdir -p tools/build/yosys/share/yosys/plugins
+  cp tools/LSOracle/core/test.ini tools/build/LSOracle/core/test.ini
```

Comment:
This is not the right way to fix this issue - if the file is required to run, then it should be fixed in the CMake build/install scripts.

In fact, when I try to run `make install PREFIX=../../build/LSOracle` I find that it is trying to install to the hardcoded path of `/usr/local/bin/lsoracle`. Please fix in the LSOracle repo so we can downstream it to the fork.

Reply:
I have already made the change upstream https://github.com/lnis-uofu/LSOracle/commit/9568a703d4aa017f55655cb863922184726a0605
The turnaround for change up stream/fork update/PR is incredibly cumbersome, so I was hoping this stopgap would resolve the issue more quickly.
As far as installation, the installation path is set using cmake's install directory variables.
```
install(TARGETS lsoracle CONFIGURATIONS RELEASE RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
install(FILES core/test.ini CONFIGURATIONS RELEASE DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/lsoracle)```
I don't know if PREFIX is what cmake is expecting, I've filed an issue to investigate https://github.com/lnis-uofu/LSOracle/issues/52

Reply:
it's not clear what copying the test.ini file into the build directory accomplishes, other than just patch a problem.  I think the yosys-plugin should be able accept the $(LSORACLE_KAHYPAR_CONFIG) as a flag rather than rewrite the script, since the plugin has other options that are not available to us now, through the plugin.
If the issue here is that lsoracle is not installed correctly, maybe a fix in the build.sh script to install it into a tools/install/LSOracle would be the better solution since cmake should handle copying of the file itself. In lsoracle it looks like it attempts to find this file relative to the executable, so that might also help eliminate the LSORACLE_KAHYPAR_CONFIG.

Reply:
I think we are getting a little off track - the config file just needs to be _somewhere_. In the build script, neither LSOracle, nor yosys, nor OpenROAD uses cmake or any other build tool to do the installation to tools/build. Everything is just a copy operation. tools/build/LSOracle was just a simple location to put it.

The lsoracle script I am generating in synth.tcl does exactly what the plugin does for the default script. I had hoped it would be a make it more obvious what the tool was doing, just like how the abc script is done below. There's no significance to the environment variable, the relevant line for calling optimization has a command argument for specifying the config file. All the path strings you see in the lsoracle code base are just defaults, you can always specify as an option to the command. The environment variable is a good idea, but outside the scope of fixing this.

Reply:
> I have already made the change upstream [lnis-uofu/LSOracle@9568a70](https://github.com/lnis-uofu/LSOracle/commit/9568a703d4aa017f55655cb863922184726a0605)

Well this repository points to a fork of [lnis-uofu/LSOracle-Plugin](https://github.com/lnis-uofu/LSOracle-Plugin), which has LSOracle as a submodule which hasn't been updated with that change. Our fork is up to date.

But in fact, the repository now looks archived saying that it is integrated into the main LSOracle repo... so there should probably be a PR to change that...

> The turnaround for change up stream/fork update/PR is incredibly cumbersome, so I was hoping this stopgap would resolve the issue more quickly.

Not really... the whole thing would probably take less than a day. There are some people on this project that are okay with stopgaps but I abhor them because this is an ongoing open-source project. If it's worth doing, then it's worth doing the right way IMO.

> As far as installation, the installation path is set using cmake's install directory variables.
> 
> ```
> install(TARGETS lsoracle CONFIGURATIONS RELEASE RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
> install(FILES core/test.ini CONFIGURATIONS RELEASE DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/lsoracle)
> ```
> 
> I don't know if PREFIX is what cmake is expecting, I've filed an issue to investigate [lnis-uofu/LSOracle#52](https://github.com/lnis-uofu/LSOracle/issues/52)

I was confusing `DESTDIR` and `PREFIX` (which is what yosys uses because it isn't CMake-based). `DESTDIR` works fine. You can close lnis-uofu/LSOracle#52

Reply:
> In the build script, neither LSOracle, nor yosys, nor OpenROAD uses cmake or any other build tool to do the installation to tools/build. Everything is just a copy operation. tools/build/LSOracle was just a simple location to put it.

This is not true: yosys is installed with `make install`. OpenROAD wasn't installed purely because of legacy cruft but #56 will fix that when I get around to it. We should move to everything being installed.

Reply:
> Well this repository points to a fork of [lnis-uofu/LSOracle-Plugin](https://github.com/lnis-uofu/LSOracle-Plugin), which has LSOracle as a submodule which hasn't been updated with that change. Our fork is up to date.
> 
Where are you seeing references to the plugin still? Everything should be completely pointed at LSOracle, the plugin was integrated into that repository specifically during the PR to get it into here in the first place..

Reply:
Not sure, I must have a very stale copy or something. In that case, you can simply update the submodule hash in this PR then. We also have a fork of [lnis-uofu/LSOracle](https://github.com/lnis-uofu/LSOracle) which is also up to date. I am going to delete the fork of LSOracle-Plugin if it is unused.

Reply:
I've updated the build to do an install rather than a copy.